src/core/hash.js

```diff
@@ -30,21 +30,22 @@ let stringify = (data) => {
  * @param {Object} sheet StyleSheet target
  * @param {Object} global Global flag
  * @param {Boolean} append Append or not
+ * @param {Boolean} keyframes Keyframes mode. The input is the keyframes body that needs to be wrapped.
  * @returns {String}
  */
-export let hash = (compiled, sheet, global, append) => {
-    // generate hash
-    let stringifiedCompiled = compiled.toLowerCase ? compiled : stringify(compiled);
+export let hash = (compiled, sheet, global, append, keyframes) => {
+    let ast = compiled[0] ? astish(compiled) : compiled;
+    let stringifiedCompiled = stringify(ast);
```

Comment:
I notice a pref deopt and I think this is the reason. `stringify`-ing all the time. Will investigate.